installers/msi-language/ActiveState/FileAssociation.cs

```diff
@@ -0,0 +1,85 @@
+﻿using Microsoft.Win32;
+using System;
+
+namespace ActiveState
+{
+
+    public class FileAssociation
+    {
+        // needed so that Explorer windows get refreshed after the registry is updated
+        [System.Runtime.InteropServices.DllImport("Shell32.dll")]
+        private static extern int SHChangeNotify(int eventId, int flags, IntPtr item1, IntPtr item2);
+
+        private const int SHCNE_ASSOCCHANGED = 0x8000000;
+        private const int SHCNF_FLUSH = 0x1000;
+
+        public string Extension { get; set; }
+        public string ProgId { get; set; }
+        public string FileTypeDescription { get; set; }
+        public string ExecutableFilePath { get; set; }
+
+        public bool SetAssociation()
+        {
+            bool madeChanges = false;
+            madeChanges |= SetKeyDefaultValue(@"Software\Classes\" + Extension, ProgId);
+            madeChanges |= SetKeyDefaultValue(@"Software\Classes\" + ProgId, FileTypeDescription);
+            madeChanges |= SetKeyDefaultValue($@"Software\Classes\{ProgId}\shell\open\command", "\"" + ExecutableFilePath + "\" \"%1\" %*");
+            return madeChanges;
+        }
+
+        public bool DeleteAssociation()
+        {
+            // We only delete the ProgId entry, as the file extension entry could be modified by other programmes
```

Comment:
We may want to confirm the default approach Windows uses when removing file type associations on uninstall. If I remember correctly it will delete the `Software\Classes\<Extension>` key as well

Reply:
I filed a story for this: https://www.pivotaltracker.com/n/projects/2203557

As there could be several bindings to `.pl` or `.wpl` (from other programmes) too, I "believe" that the MSI installer logic is actually smarter here, and would somehow restore a still existing installation's binding.